modules/transport/xmlrpc/main.c

```diff
@@ -511,49 +514,68 @@ xmlrpcmethod_metadata(void *conn, int parc, char *parv[])
 		}
 	}
 
-	if (parc < 2)
+	if (parc < 4)
 	{
 		xmlrpc_generic_error(fault_needmoreparams, "Insufficient parameters.");
 		return 0;
 	}
 
-	if (*parv[0] == '#')
+	if (*parv[1] != '\0' && strlen(parv[0]) > 1)
```

Comment:
The `parc` check above already guarantees that you have at least 4 parameters. This entire check is unnecessary.